File: HAI704I - Architectures Distribuées/TP2/TP2-Server/TP2-ServiceServer/model/reservation/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TP2_Server.model.customer;
using TP2_Server.model.room;

namespace TP2_Server.model.reservation
{
    class Reservation
    {
        public DateTime Start
        {
            get;
        }

        public DateTime End
        {
            get;
        }

        private Customer customer;
        private Room room;

        public Reservation(DateTime start, DateTime end, Customer customer, Room room)
        {
            this.Start = start;
            this.End = end;
            this.customer = customer;
            this.room = room;
        }

        public bool Concern(Room room)
        {
            return this.room.Equals(room);
        }

        public bool Intersect(DateTime start, DateTime end)
        {
            return !(end < this.Start || start > this.End);
        }

        public bool IsLaterThan(DateTime date)
        {
            return date < this.Start;
        }

        public bool IsEarlierThan(DateTime date)
        {
            return date > this.End;
        }
    }
}

File: HAI704I - Architectures Distribuées/TP3/TP3-Server/src/main/java/m1/hotels/model/reservation/Reservation.java
package m1.hotels.model.reservation;

import java.util.Date;

import m1.hotels.model.customer.Customer;
import m1.hotels.model.room.Room;

public class Reservation
    {
		private static int MAX_ID = 0;
		
		private final int id;
		private Date start;
        private Date end;
        private Customer customer;
        private Room room;

        public Reservation(Date start, Date end, Customer customer, Room room)
        {
        	this.id = Reservation.MAX_ID++;
            this.start = start;
            this.end = end;
            this.customer = customer;
            this.room = room;
        }

        public int getId()
        {
        	return this.id;
        }
        
        public boolean concern(Room room)
        {
            return this.room.equals(room);
        }

        public boolean intersect(Date start, Date end)
        {
            return !(end.before(this.start) || start.after(this.end));
        }

        public boolean isLaterThan(Date date)
        {
            return date.before(this.start);
        }

        public boolean isEarlierThan(Date date)
        {
            return date.after(this.end);
        }

		public Date getStart()
		{
			return start;
		}

		public Date getEnd()
		{
			return end;
		}       
    }
